core/shared/src/main/scala/zio/random/Random.scala

```diff
@@ -80,4 +82,21 @@ object Random extends Serializable {
           }
       buffer <- bufferRef.get
     } yield buffer.toList
+
+  def nextLongWith(nextLong: UIO[Long], n: Long): UIO[Long] =
```

Comment:
Instead of having this helper function, you could make this the default on the `Random.Service` trait. Otherwise it will appear in the `Random` object and users may wonder what it is for.

How does that sound?

Otherwise, looks perfect and ready to merge!

Reply:
Oh shoot, I forgot to make that `protected[zio]` like the `shuffleWith` helper method above. Does that work?

Reply:
Yes, that works!